### ISSUE TITLE
[FR] autocomplete for links in insert mode

### ISSUE THREAD

**Issue by niderhoff** [nvim-telekasten/telekasten.nvim#287]
**Please confirm**
- [x] I am running the latest version of this plugin

**Is your feature request related to a problem? Please describe.**
When inserting new links, it would be great instead of telescope picker to have a autocomplete popup, (maybe nvim-cmp integration?) for links. 

**Describe the solution you'd like**
If I type [[ in insert mode, I want to have a popup with all the notes in the current vault.

<img width="529" alt="image" src="https://github.com/renerocksai/telekasten.nvim/assets/5991543/d0d5e485-2cc0-4a56-b4d3-92861121031a">

**Describe alternatives you've considered**
current insert_link feature

**Additional context**
If we add a nvim-cmp integration, I think it's pretty easy to integrate with cmp so that on typing `[[` it will only show telekasten related autocompletion.

**Comment by lambtho12** [comment]
This looks like a good idea, yes. But I have absolutely no idea on how to do it. Wouldn't we need a LSP for that? There are already some issues about adding a LSP for telekasten.

**Comment by niderhoff** [comment]
Yeah the [cmp documentation](https://github.com/knubie/nvim-cmp#create-a-custom-source) is not really perfect in that regard. But probably it involves keeping an updated list somewhere in the background and providing it to cmp when it asks for that. If the interface needs to be LSP is not clear to me, but its documentation is mentioned in the aforementioned documentation as a "must read", so...
